cs/Serializer/VowpalWabbitSerializer.cs

```diff
@@ -137,13 +181,32 @@ private void Dispose(bool disposing)
             }
         }
 
-		public void ReturnExampleToPool(VowpalWabbitExample ex)
+        /// <summary>
+        /// Accepts an example back into this pool.
+        /// </summary>
+        /// <param name="example">The example to be returned.</param>
+		public void ReturnExampleToPool(VowpalWabbitExample example)
         {
             if (this.exampleCache == null)
             {
                 throw new ObjectDisposedException("VowpalWabbitSerializer");
             }
 
+#if DEBUG
+            CacheEntry cacheEntry;
+            if (!this.reverseLookup.TryGetValue(example, out cacheEntry))
+            {
+                throw new ArgumentException("Example is not found in pool");
+            }
+
+            if (!cacheEntry.InUse)
```

Comment:
CacheEntries are only added a few lines below. Never null.

From: Luong Hoang [mailto:notifications@github.com]
Sent: Monday, September 14, 2015 2:54 PM
To: JohnLangford/vowpal_wabbit <vowpal_wabbit@noreply.github.com>
Cc: Markus Cozowicz <marcozo@microsoft.com>
Subject: Re: [vowpal_wabbit] Documentation (#754)


In cs/Serializer/VowpalWabbitSerializer.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fJohnLangford%2fvowpal_wabbit%2fpull%2f754%23discussion_r39432017&data=01%7c01%7cmarcozo%40064d.mgd.microsoft.com%7cef5d7e5d95634b7f0f8308d2bd35e294%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=5bPZtLhPDMMNC6B4gyVYmgRWXdmqiY%2fIbj6ZkNkgPBQ%3d>:

>          {

>              if (this.exampleCache == null)

>              {

>                  throw new ObjectDisposedException("VowpalWabbitSerializer");

>              }

>

> +#if DEBUG

> +            CacheEntry cacheEntry;

> +            if (!this.reverseLookup.TryGetValue(example, out cacheEntry))

> +            {

> +                throw new ArgumentException("Example is not found in pool");

> +            }

> +

> +            if (!cacheEntry.InUse)

can cacheEntry be null here?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fJohnLangford%2fvowpal_wabbit%2fpull%2f754%2ffiles%23r39432017&data=01%7c01%7cmarcozo%40064d.mgd.microsoft.com%7cef5d7e5d95634b7f0f8308d2bd35e294%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=VNfYEeHPeeo4iw%2fIaivHPF3c7j0Ntcbg4todamFLn%2bs%3d>.